src/Core/Grid/Column/Type/Common/ToggleColumn.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Column\Type\Common;
+
+use PrestaShop\PrestaShop\Core\Grid\Column\AbstractColumn;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This Column is used to display booleans.
+ * - it will display an icon instead of the value
+ * - if user clicks on it, this triggers a toggle of the boolean value
+ */
+final class ToggleColumn extends AbstractColumn
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getType()
+    {
+        return 'toggle';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setRequired([
+                'field_to_display', // field which is the boolean to display
+                'field_to_use_as_id', // field which can be used as ID to perform the toggle
+                'url_toggle_callback', // URL called when user clicks on togglable row
```

Comment:
this one can also be simplified, maybe `toggle_url`? :) 

p.s. can it be route name instead? so you can generate url in template. :)

Reply:
even easier => url

Reply:
@sarjon I can dot it but it would not be consistent with other [LogGridDefinitionFactory columns](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php#L50). Is that OK ?

Reply:
this doesn't have to be as GridDefinition factories and Column are different things, you may compare with others Column types. As naming is quite hard, let's give us time for now with the chose to update it until the release of 1.7.5, ok? :)

Reply:
It's not blocking, we can merge it with this naming already :+1: